### DISCUSSION TITLE
WPEWebProcess memory handle

### DISCUSSION THREAD

**Discussion by gnulux** [Igalia/cog#724] [Q&A]
Hi, 

I'm wondering if there is a way to limit memory requested by WPEWebProcess ? 
I'm running COG on embedded systems where memory size is short . 
Do variables like WPE_RAM_SIZE , WPE_POLL_MAX_MEMORY have an impact on memory consomption ?

**Comment by magomez** [comment]
You don't specify which version of WPE you're running, or from which repository.

From WPE 2.38 on you have the MemoryPressureSettings mechanism (https://github.com/WebKit/WebKit/blob/main/Source/WebKit/UIProcess/API/glib/WebKitMemoryPressureSettings.cpp). This API allows setting a memory limit and some percentages of it, so WPEWebProcess will try (when possible) to stay under the limit set.

If you're using WPE from https://github.com/WebPlatformForEmbedded/WPEWebKit, then (on 2.38 as well) WPE_RAM_SIZE can be used to define value for WPE to use as the total RAM size of the device. This is used to decide the size of several caches inside WPE, so setting it to smaller values (down to a limit) will reduce a bit the memory footprint.

WPE_POLL_MAX_MEMORY was used before 2.38, before the introduction of the memory pressure settings.

**Reply by gnulux** [reply]
Sorry you're right 

I'm running cog 0.12.3 (WPE WebKit 2.34.6) from Yocto Hardknott . 
meta-webkit seems to gather the sources package from https://wpewebkit.org/releases 
I tried :  WPE_RAM_SIZE="128m" WPE_POLL_MAX_MEMORY="databaseprocess:5m,networkprocess:5m,webprocess:10m,rpcprocess:5m"

But the memory is still moving from 180 to 400M. I would like to fix the memory to max: 200M or that WPEWebProcess release memory when on a static page.

**Comment by magomez** [comment]
If you're getting the wpe sources from https://wpewebkit.org/releases then the env vars you're trying to use are not defined. Those are only defined in the versions from https://github.com/WebPlatformForEmbedded/WPEWebKit. You need to either switch to that repository or port the patches that add the env vars to the source you're using.

**Reply by gnulux** [reply]
Unfortunatly 2.38 (downstream) does not build from my cross-compiled environment because of gi-docgen package.

How can i find in dowstream repository the latest commit related to upstream version 2.34.6?

**Reply by magomez** [reply]
You can remove the need to have gi-docgen by disabling introspection  in Source/cmake/OptionsWPE.cmake.

Unfortunately not all releases of WPE are in the WebPlatformForEmbedded repository. It's either 2.28 or 2.38.

**Reply by gnulux** [reply]
Nice, i succeed to build 2.38 , it works better than the previous version.
Memory consumption is  flat regarding older graphic and i did not set env parameters for now.

Before
![image](https://github.com/user-attachments/assets/93dbb1f2-1b5a-4c34-958c-46664336886f)

After
![image](https://github.com/user-attachments/assets/6faa437b-84ba-491d-b4fc-9dd9a8d18a4a)

I will try with parameters. 
Many Thanks for your advises
